crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -1180,6 +1181,252 @@ impl<'w> EntityWorldMut<'w> {
         self
     }
 
+    /// Removes all components of this [`Bundle`] and its required components that are not required by other components of this entity.
+    ///
+    /// This function can be noticeably slower than simple remove or retain functions because it dynamically determines which components
+    /// are still required by entity components outside of the [`Bundle`].
```

Comment:
Given this observation, would it make sense to cache this transition in the archetype graph?

Reply:
I probably should do some time measurements.... For a small number of components, the removal time will not be meaningfully increased, but caching will significant increase memory consumption